window.odin

```diff
@@ -1,197 +1,66 @@
-    class := win32.register_class_ex_a(&window_class);
-    if class == 0 {
-        logln("Error in register_class_ex_a(): ", get_and_clear_last_win32_error());
-        return {}, false;
-    }
+    window_class.class_name = cstring("WindowClass");
 
-    //
-    window: Window;
-    currently_updating_window = &window;
+    class := win32.register_class_ex_a(&window_class);
+    assert (class != 0);
 
-    rect := win32.Rect{0, 0, cast(i32)width, cast(i32)height};
+    rect := win32.Rect{ 0, 0, cast(i32)width, cast(i32)height };
     win32.adjust_window_rect(&rect, win32.WS_OVERLAPPEDWINDOW, false);
-    win32.create_window_ex_a(0,
-                             window_class.class_name,
-                             strings.clone_to_cstring(name),
-                             win32.WS_OVERLAPPEDWINDOW | win32.WS_VISIBLE,
-                             200, // todo(josh): get the resolution of the monitor and put the window in the center
-                             100, // todo(josh): get the resolution of the monitor and put the window in the center
-                             rect.right - rect.left,
-                             rect.bottom - rect.top,
-                             nil,
-                             nil,
-                             window_class.instance,
-                             nil);
-    
-
-    if window.platform_data.window_handle == nil {
-   		logln(win32.get_last_error());
-   		return {}, false;
-    }
 
-    return window, true;
+    window_handle := 
+        win32.create_window_ex_a(
+            0,
+            window_class.class_name,
+            strings.clone_to_cstring(name),
+            win32.WS_OVERLAPPEDWINDOW | win32.WS_VISIBLE,
+            200,
+            100,
+            rect.right - rect.left,
+            rect.bottom - rect.top,
+            nil,
+            nil,
+            window_class.instance,
+            nil);
+
+    assert(window_handle != nil);
+    return window_handle;
 }
 
-g_context: runtime.Context;
-
-wnd_proc :: proc "std" (window_handle: win32.Hwnd, message: u32, wparam: win32.Wparam, lparam: win32.Lparam) -> win32.Lresult {
-    context = g_context;
-
-    assert(currently_updating_window != nil);
+wnd_proc :: proc "std" (
+    window_handle: win32.Hwnd,
+    message: u32,
+    wparam: win32.Wparam,
+    lparam: win32.Lparam) -> win32.Lresult {
 
     result: win32.Lresult;
 
-    @static mouse_capture_sum: int;
-
     switch (message) {
-        case win32.WM_CREATE: {
-            defer LOG_WINDOWS_ERROR();
-            assert(currently_updating_window.platform_data.window_handle == nil);
-            currently_updating_window.platform_data.window_handle = window_handle;
-
-            assert(currently_updating_window.platform_data.device_context == nil);
-            dc := win32.get_dc(window_handle);
-            currently_updating_window.platform_data.device_context = dc;
-
-            // todo(josh): look into what this stuff means
```

Comment:
FYI this is for creating an opengl context. Given that this is a directX binding, you probably don't need/want this.